src/routes/task/task.routes.ts

```diff
@@ -0,0 +1,94 @@
+import {Router} from 'express'
+import Task from './taskModel'
+
+const router = Router()
+
+router.post('/api/tasks', async(req, res) => {
+	try {
+		const {text} = req.body
+		if( /^\s*$/.test(text)) {
+			return res.status(400).json({message: 'Text is empty!'})
+		}
+		const data = new Task({
+			text: req.body.text,
+			date: new Date().toLocaleString(),
+			description: '',
+			taskStatus: 'todo',
+			userId: req.user._id
+		})
+		const dataToSave = data.save()
+		res.status(200).json(dataToSave)
+	}
+	catch(error) {
+		res.status(400).json({message: error.message})
+	}
+})
+router.get('/api/tasks', async(req, res) => {
+	try {
+		const data = await Task.find({userId: req.user._id})
+		if(data) {
+			res.json(data)
+		} else {
+			res.status(404).send({message: 'Tasks do not exists!'})
+		}
+	}
+	catch(error){
+		res.status(500).json({message: error.message})
+	}
+})
+router.delete('/api/tasks/:id', async(req, res) => {
+	try {
+		const {id} = req.params
+		const data = await Task.findById(req.params.id)
+		if(!data) {
+			res.status(404).send({message: 'Task does not exist!'})
+		}
+		if(data.userId === req.user._id) {
+			await Task.findByIdAndDelete(id)
+			res.send()
+		} else {
+			res.status(403).send({message: 'Task does not belong to user'})
+		}
+	}
+	catch (error) {
+		res.status(405).json({message: error.message})
+	}
+})
+router.patch('/api/tasks/:id', async(req, res) => {
+	try {
+		const updatedData = req.body
+		const {id} = req.params
+		const data = await Task.findById(req.params.id)
+		if(!data) {
+			res.status(404).send({message: 'Task does not exist!'})
+		}
+		const taskDescription = await Task.findById(req.params.id)

```

Comment:
Po znów odpytujesz bazę skoro powyżej już to zrobiłeś?

Reply:
zmienione